radix-engine-interface/src/wasm/input.rs

```diff
@@ -0,0 +1,394 @@
+use crate::api::types::*;
+use crate::scrypto;
+use radix_engine_interface::api::api::SysInvokableNative;
+use radix_engine_interface::data::IndexedScryptoValue;
+use sbor::rust::fmt::Debug;
+use sbor::rust::string::String;
+use sbor::rust::vec::Vec;
+
+#[derive(Debug)]
+#[scrypto(TypeId, Encode, Decode)]
+pub enum RadixEngineInput {
+    InvokeScryptoFunction(ScryptoFunctionIdent, Vec<u8>),
+    InvokeScryptoMethod(ScryptoMethodIdent, Vec<u8>),
+    InvokeNativeFn(NativeFnInvocation),
+
+    CreateNode(ScryptoRENode),
+    GetVisibleNodeIds(),
+    DropNode(RENodeId),
+
+    LockSubstate(RENodeId, SubstateOffset, bool),
+    DropLock(LockHandle),
+    Read(LockHandle),
+    Write(LockHandle, Vec<u8>),
+
+    GetActor(),
+    EmitLog(Level, String),
+    GenerateUuid(),
+    GetTransactionHash(),
+}
+
+#[derive(Debug)]
+#[scrypto(TypeId, Encode, Decode)]
+pub enum NativeFnInvocation {
+    Method(NativeMethodInvocation),
+    Function(NativeFunctionInvocation),
+}
+
+#[derive(Debug)]
+#[scrypto(TypeId, Encode, Decode)]
+pub enum NativeMethodInvocation {
+    AccessRules(AccessRulesMethodInvocation),
+    Metadata(MetadataMethodInvocation),
```

Comment:
I think, at some point, we will need to limit and isolate these "any" methods. My inclination is to make it:
```rust
pub enum NativeMethodInvocation {
    Global(GlobalMethodInvocation)
}
```